packages/cli/src/base-generator.ts

```diff
@@ -0,0 +1,132 @@
+        const indexFileContent = this.generateIndexSource(indexFileTargetPath);
+        ensureDirectory(fullOutDir, fs);
+        tryRun(
+            () => fs.writeFileSync(indexFileTargetPath, '\n' + indexFileContent + '\n'),
+            'Write Index File Error'
+        );
+        this.log('[Generator Index]', 'creating index file: ' + indexFileTargetPath);
+    }
+
+    public filename2varname(filePath: string) {
+        const varname = basename(basename(filePath, '.css'), '.st') // remove prefixes and .st.css ext
+            .replace(/^\d+/, ''); // remove leading numbers
+        return upperfirst(camelcase(varname));
+    }
+
+    protected generateIndexSource(_indexFileTargetPath: string) {
+        return this.indexFileOutput
+            .map((_) => createImportForComponent(_.from, _.reExports))
+            .join('\n');
+    }
+
+    private checkForCollisions(reExports: ReExports, filePath: string) {
+        this.collisionDetector.detect(reExports.root, filePath);
+
+        for (const asName of Object.values(reExports.parts)) {
+            this.collisionDetector.detect(asName, filePath);
+        }
+
+        for (const asName of Object.values(reExports.vars)) {
+            this.collisionDetector.detect(asName, filePath);
+        }
+
+        for (const asName of Object.values(reExports.stVars)) {
+            this.collisionDetector.detect(asName, filePath);
+        }
+
+        for (const asName of Object.values(reExports.keyframes)) {
+            this.collisionDetector.detect(`keyframes(${asName})`, filePath);
+        }
+
+        if (this.collisionDetector.collisions.size) {
+            let errorMessage = 'Name Collision Error:';
+            for (const [name, origin] of this.collisionDetector.collisions) {
+                errorMessage += `\nexport symbol ${name} from ${filePath} is already used by ${origin}`;
+            }
+            throw new Error(errorMessage);
+        }
+    }
+}
+
+class NameCollisionDetector<Origin> {
+    nameMapping = new Map<string, Origin>();
+    collisions = new Map<string, Origin>();
+    detect(name: string, origin: Origin) {
+        if (this.nameMapping.has(name)) {
+            this.collisions.set(name, this.nameMapping.get(name)!);
+        } else {
+            this.nameMapping.set(name, origin);
+        }
+    }
+}
+
+function createImportForComponent(from: string, reExports: ReExports) {
+    const namedPart = [
+        ...Object.entries(reExports.parts).map(symbolMapper),
+        ...Object.entries(reExports.stVars).map(symbolMapper),
+        ...Object.entries(reExports.vars).map(symbolMapper),
+        ...Object.entries(reExports.keyframes).map(keyframesSymbolMapper),
+    ].join(', ');
+
+    const usagePart = Object.values(reExports.parts)
+        .map((exportName) => `.root .${exportName}{}`)
+        .join(' ');
+
+    return `:import {-st-from: ${JSON.stringify(from)};-st-default:${reExports.root};${
```

Comment:
When we introduce `@st-import`, should we update the syntax here?

Reply:
not really. but we can.